sotodlib/tod_ops/fft_ops.py

```diff
@@ -120,9 +121,60 @@ def rfft(
     freqs = np.fft.rfftfreq(n, delta_t)
 
     if other_idx is not None and other_idx != 0:
-        return b.transpose(), freqs
+        return rfft.b.transpose(), freqs
+
+    return rfft.b, freqs
+
+
+@dataclass
+class RFFTObj:
+    """
+    Dataclass to store information needed for rfft.
+
+    Attributes:
 
-    return b, freqs
+        n_det: Number of detectors this object was built for.
+
+        n: Number of samples this object was built for.
+
+        a: Buffer for the real part of the FFT.
+
+        b: Buffer for the complex part of the FFT.
+
+        _t_forward: Function for performing the forward FFT.
+                    This is None if this direction wasn't requested.
+                    The t_forward property wraps this in with check that the
+                    attribute isn't None.
+
+        _t_backward: Function for performing the backward FFT.
+                     This is None if this direction wasn't requested.
+                    The t_backward property wraps this in with check that the
+                    attribute isn't None.
+    """
+    n_det: int
+    n: int
+    a: NDArray[np.float32]
+    b: NDArray[np.complex64]
+    _t_forward: Optional[Callable]
+    _t_backward: Optional[Callable]
+
+    def validate(self, shape):
+        if shape != (self.n_det, self.n):
+            raise ValueError("Data is wrong shape for the rfft object")
+        if self._t_forward is None and self._t_backward is None:
+            raise ValueError("Neither t_forward nor t_forward specified!")
+
+    @property
+    def t_forward(self):
```

Comment:
Alternately you could have a default implementation of the t_* that just raises an error (i.e. slightly more informative than attribute not found). Then no need for private members.